extensions/stackdriver/stackdriver.cc

```diff
@@ -206,6 +221,78 @@ flatbuffers::DetachedBuffer getLocalNodeMetadata() {
   return ::Wasm::Common::extractNodeFlatBufferFromStruct(node);
 }
 
+void fillAuthzDryRunInfo(
+    std::unordered_map<std::string, std::string>& extra_labels) {
+  auto md = getProperty({"metadata", "filter_metadata", kRBACHttpFilterName});
+  if (!md.has_value()) {
+    md = getProperty({"metadata", "filter_metadata", kRBACNetworkFilterName});
+    if (!md.has_value()) {
+      LOG_DEBUG("RBAC metadata not found");
+      return;
+    }
+  }
+
+  bool shadow_deny_result = false;
+  bool shadow_allow_result = false;
+  bool has_shadow_metadata = false;
+  std::string shadow_deny_policy = "";
+  std::string shadow_allow_policy = "";
+  for (const auto& [key, val] : md.value()->pairs()) {
+    LOG_DEBUG(absl::StrCat("RBAC metadata found: key=", key, ", value=", val));
+    if (key == kDryRunDenyShadowEngineResult) {
+      shadow_deny_result = (val == "allowed");
+    } else if (key == kDryRunAllowShadowEngineResult) {
+      shadow_allow_result = (val == "allowed");
+    } else if (key == kDryRunDenyShadowEffectiveId) {
+      shadow_deny_policy = val;
+    } else if (key == kDryRunAllowShadowEffectiveId) {
+      shadow_allow_policy = val;
+    } else {
+      continue;
+    }
+    has_shadow_metadata = true;
+  }
+
+  if (!has_shadow_metadata) {
+    LOG_DEBUG("RBAC dry-run metadata not found");
+    return;
+  }
+
+  LOG_DEBUG("RBAC dry-run result found");
+  bool shadow_result = false;
+  std::string shadow_effective_policy = "";
+  if (shadow_deny_result && shadow_allow_result) {
+    // If allowed by both DENY and ALLOW policy, the final shadow_result should
+    // be true (allow) and the shadow_effective_policy should be from the ALLOW
+    // policy.
+    shadow_result = true;
+    shadow_effective_policy = shadow_allow_policy;
+  } else {
+    // If denied by either DENY or ALLOW policy, the final shadow_reulst should
+    // be false (denied).
+    shadow_result = false;
+    if (!shadow_deny_result) {
+      // If denied by DENY policy, the shadow_effective_policy should be from
+      // the DENY policy.
+      shadow_effective_policy = shadow_deny_policy;
+    } else {
+      // If denied by ALLOW policy, the shadow_effective_policy shold be from
+      // the ALLOW policy.
+      shadow_effective_policy = shadow_allow_policy;
+    }
+  }
+
+  extra_labels["dry_run_result"] =
+      shadow_result ? kRbacAccessAllowed : kRbacAccessDenied;
+  std::string policy_name, policy_namespace, policy_rule_index;
+  if (RE2::PartialMatch(shadow_effective_policy, authz_name_match,
```

Comment:
I am a bit worried about the performance impact of this. Is it possible to do a customized parsing instead of regex?

Also could you add an example policy name in the comment?

Reply:
I think it should be fine as the same regex is already used in the logger.cc for authz deny logging parsing. I updated to not to use regex anyway as it's a simple change, I can update the usage in logger.cc in a follow up PR if needed.